### PR TITLE
Send md5 of customer key

### PR DESCRIPTION
**Pull request by robertstrache**
This PR should send the MD5 checksum of a customer provided encryption key.

It is taken from a comment of another user in https://github.com/vmware-tanzu/velero/issues/7693
I have just turned it into a PR since it seems to solve our issue described in
https://github.com/vmware-tanzu/velero/issues/7693

### CODE REVIEW

**Review by reasonerjt — DISMISSED** [review]
lgtm

File: velero-plugin-for-aws/object_store.go [hunk 1]
```diff
@@ -267,6 +282,7 @@ func (o *ObjectStore) PutObject(bucket, key string, body io.Reader) error {
 	case o.sseCustomerKey != "":
 		input.SSECustomerAlgorithm = aws.String("AES256")
 		input.SSECustomerKey = &o.sseCustomerKey
+		input.SSECustomerKeyMD5 = &o.sseCustomerKeyMd5
```
Comment (reasonerjt) [inline velero-plugin-for-aws/object_store.go:285]: Will the func "ObjectExists" also need this change?

**Comment by gschei** [comment]
This PR has some code duplication (readCustomerKey is called twice now), also comment was correct, that this is missing for head and get operations. I created a new PR for this: https://github.com/vmware-tanzu/velero-plugin-for-aws/pull/225

**Comment by reasonerjt** [comment]
Thanks @gschei I'm closing this one.

**Pull request closed without merging**